pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,465 @@
+
+    .. code-block:: python
+
+        plt.style.use('Solarize_Light2')
+        drawer = MPLDrawer(n_wires=5,n_layers=5)
+
+        drawer.label(["0","a",r"$|\Psi\rangle$",r"$|\theta\rangle$", "aux"])
+
+        drawer.box_gate(0, [0,1,2,3,4], "Entangling Layers", rotate_text=True)
+        drawer.box_gate(1, [0, 1], "U(θ)")
+        drawer.box_gate(1, 4, "X", color='lightcoral')
+
+        drawer.SWAP(1, (2, 3))
+        drawer.CNOT(2, (0,2), color='forestgreen')
+
+        drawer.ctrl(3, [1,3])
+        drawer.box_gate(3, 2, "H", zorder_base=2)
+
+        drawer.ctrl(4, [1,2])
+
+        drawer.measure(5, 0)
+
+        drawer.fig.suptitle('My Circuit', fontsize='xx-large')
+
+    .. figure:: ../../_static/drawer/example_Solarize_Light2.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+
+    """
+
+    def __init__(self, n_layers, n_wires, figsize=None):
+
+        self.n_layers = n_layers
+        self.n_wires = n_wires
+
+        # half the width of a box
+        # is difference between center and edge
+        self._box_dx = 0.4
+
+        self._circ_rad = 0.3
+        self._ctrl_rad = 0.1
+        self._swap_dx = 0.2
+
+        ## Creating figure and ax
+
+        if figsize is None:
+            figsize = (self.n_layers + 3, self.n_wires + 1)
+
+        self.fig = plt.figure(figsize=figsize)
+        self.ax = self.fig.add_axes(
+            [0, 0, 1, 1],
+            xlim=(-2, self.n_layers + 1),
+            ylim=(-1, self.n_wires),
+            xticks=[],
+            yticks=[],
+        )
+
+        # adding wire lines
+        for wire in range(self.n_wires):
+            line = plt.Line2D((-1, self.n_layers), (wire, wire), zorder=1)
+            self.ax.add_line(line)
+
+    def label(self, labels):
+        """Label each wire.
+
+        Args:
+            labels [Iterable[str]]: Iterable of labels for the wires
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=1)
+            drawer.label(["a", "b"])
+
+        .. figure:: ../../_static/drawer/labels.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        for wire, ii_label in enumerate(labels):
+            self.ax.text(-1.5, wire, ii_label)
+
+    def box_gate(
+        self, layer, wires, text="", extra_width=0, rotate_text=False, zorder_base=0, color=None
+    ):
+        """Draws a box and adds label text to it's center.
+
+        Args:
+            layer (Int)
+            wires (Union[Int, Iterable[Int]])
+            text (str)
+
+        Kwargs:
+            extra_width=0 (float): Extra box width
+            rotate_text=False (Bool): whether to rotate text 90 degrees. Helpful to long labels and
+                multi-wire boxes.
+            zorder_base (Int): shift the object in zorder
+            color=None: mpl compatible color designation
```

Comment:
```suggestion
            extra_width (float): Extra box width
            rotate_tex t(Bool): whether to rotate text 90 degrees. Helpful to long labels and
                multi-wire boxes.
            zorder_base (Int): shift the object in zorder
            color (None or str): mpl compatible color designation
```

There is a rendering issue in the docs, I think this would solve it.